### PR TITLE
WSTEAM1-760: Alter embed semantics

### PR DESCRIPTION
**Pull request by shayneahchoon**
Resolves [WSTEAM1-760
](https://jira.dev.bbc.co.uk/browse/WSTEAM1-760)

Overall changes
======
Changes semantics of embeds, so that they do not appear as `<Figure />`. 

Code changes
======
- _`src/app/legacy/psammead/psammead-social-embed/src/index.jsx` changed to create a unique id for the warning label._
- `/src/app/legacy/psammead/psammead-social-embed/src/SkipLinkWrapper/index.jsx` changed to include an `aria-describedby` property, which refers to the id of the warning label.

Testing
======
1. _Visit: [https://www.test.bbc.com/pidgin/live/c7p765ynk9qt](https://www.test.bbc.com/pidgin/live/c7p765ynk9qt)._
2. _Check screenreader UX for embeds, specifically check that embeds are NOT described as Figure._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

**Comment by greenc05** [comment]
Thanks @shayneahchoon May I check did you intend for the id value to be so alien? id=":r0:"

Also think we have lost some spacing around the warning text, as the small tag is an inline element as apposed to a figure/figcation which is a block level element.

**Comment by shayneahchoon** [comment]
> Thanks @shayneahchoon May I check did you intend for the id value to be so alien? id=":r0:"
> 
> Also think we have lost some spacing around the warning text, as the small tag is an inline element as apposed to a figure/figcation which is a block level element.

Ello, the id values are actually automated as recommended by React for aria attributes: https://react.dev/reference/react/useId

File: src/app/legacy/containers/SocialEmbed/__snapshots__/index.test.jsx.snap [hunk 1]
```diff
@@ -403,15 +405,16 @@ exports[`SocialEmbedContainer AMP should match snapshot 1`] = `
             Accept and continue
           </button>
         </div>
-        <figcaption
+        <small
           class="emotion-14 emotion-15"
+          id=":r8:"
         >
           <span>
```
Comment (greenc05) [inline src/app/legacy/containers/SocialEmbed/__snapshots__/index.test.jsx.snap]: This span shouldn't be output, this is for captions

**Comment by greenc05** [comment]
Subject to any front end changes this is a pass for our supported screen readers

**Mobile**
- [x] VoiceOver / Safari: As expected
- [x] Talkback / Chrome: As expected

**Desktop**
- [x] VoiceOver / Safari: VoiceOver is not automatically reading out aria-describedby but instead it tells you about the "more" menu (Control + Option + Command + "/"). If you open this menu the warning text is available to read. This looks like a new feature for VoiceOver Mac https://support.google.com/chrome/thread/206932536/aria-describedby-behaviour-is-updat[…]he-latest-chrome-version-111-0-5563-64-official-build?hl=en](https://support.google.com/chrome/thread/206932536/aria-describedby-behaviour-is-updat[%E2%80%A6]he-latest-chrome-version-111-0-5563-64-official-build?hl=en 
- [x] Jaws / Chrome / Edge: With regards to aria-describedby: Didn’t read when navigating via the arrow keys. Works as expected when you navigate by links. [Looks like a known bug with Jaws](https://github.com/FreedomScientific/standards-support/issues/719).
- [x] NVDA / Chrome / Firefox: With regards to aria-describedby: Didn’t read when navigating via the arrow keys. Works as expected when you navigate by links. [Looks like a known bug / feature of NVDA](https://github.com/FreedomScientific/standards-support/issues/719).

File: src/app/legacy/psammead/psammead-social-embed/src/CaptionWrapper/index.jsx [hunk 2]
```diff
@@ -34,40 +35,40 @@ const FigCaption = styled.figcaption`
...
+  descById: null,
 };
 
 CaptionWrapper.propTypes = {
   children: node.isRequired,
   service: string.isRequired,
-  textPrefixVisuallyHidden: string,
   text: string.isRequired,
   additionalText: string,
+  descById: string,
```
Comment (karinathomasbbc) [inline src/app/legacy/psammead/psammead-social-embed/src/CaptionWrapper/index.jsx]: Could we use a more meaningful variable name here?

```suggestion
  describedById: string,
```

**Review by karinathomasbbc — APPROVED** [review]

**Review by Isabella-Mitchell — APPROVED** [review]
Agree with Karina's comment on the variable name. Otherwise looks good.

**Review by HarveyPeachey — APPROVED** [review]
Looks good, I agree with Karina though `descById` is hard to understand

**Pull request merged**